test/system/src/test/java/io/pravega/test/system/PravegaTest.java

```diff
@@ -154,13 +157,16 @@ public void simpleTest() throws InterruptedException {
                 String event = reader.readNextEvent(6000).getEvent();
                 if (event != null) {
                     log.debug("Read event: {} ", event);
+                    readCount++;
                 }
             } catch (ReinitializationRequiredException e) {
                 log.error("Unexpected request to reinitialize {}", e);
                 System.exit(0);
             }
         }
+        assertEquals(writeCount, readCount);
```

Comment:
This is only checking that the read count is `NUM_EVENTS`, I don't think we even need the `writeCount` in the way this patch is structured. At the same time, I wonder whether `writeCount` can be different from `NUM_EVENTS`. For example, in some crash scenarios, is it possible that the write count is smaller than `NUM_EVENTS` because retries exhaust but we proceed with the test case?